### PR TITLE
feat: Added pagination of users

### PR DESCRIPTION
**Pull request by metodej**
* Added to new method getUsersPage which returns paginated users
based on UsersPageQuery.
* UsersPageQuery consist of page size, offset, order, sort column and
possibly search string and withoutVo option. Search string filters
users to return only those with given string as name or other
aspect we search by. If withoutVo is set true, the method returns
only users that are not in any vo.

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java [hunk 1]
```diff
@@ -1406,4 +1456,37 @@ public void deleteSponsorLinks(PerunSession sess, User sponsor) {
...
+				" ,count(*) OVER() AS total_count" +
+				" FROM users";
+
+		String selectWithMembers =
+			"SELECT " + userMappingSelectQuery +
+				" ,count(*) OVER() AS total_count" +
+				" FROM users LEFT JOIN members on members.user_id = users.id";
+
+
+		return !query.isWithoutVo() && Objects.equals(query.getSearchString(), "") ? select : selectWithMembers;
```
Comment (cuadradek) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java:1471]: We can use `isEmpty(query.getSearchString()))` here as well.

**Review by Johaney-s — APPROVED** [review]
Nice work, everything look good to me.

**Review by cuadradek — COMMENTED** [review]
Looking good, I have only couple of nit-picks.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java [hunk 2]
```diff
@@ -902,6 +904,26 @@ public UserExtSource getUserExtSourceByExtLogin(PerunSession sess, ExtSource sou
...
+		List<AttributeDefinition> attrDefs = new ArrayList<>();
+		for (String attrsName : attrNames) {
+			try {
+				attrDefs.add(getPerunBl().getAttributesManagerBl().getAttributeDefinition(sess, attrsName));
+			} catch (AttributeNotExistsException e) {
+				//pass
+			}
+		}
+
+		richUsers = convertUsersToRichUsersWithAttributes(sess, richUsers, attrDefs);
```
[the remark this answers is not in the document]
Comment (metodej) [reply perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java]: Nice, that looks better, thanks.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java (hunk 1 above)
Comment (metodej) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java]: Fixed.

**Review by cuadradek — APPROVED** [review]

**Pull request merged**